cmd/clusterctl/client/multicluster_discovery.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MultiClusterTree struct {
```

Comment:
I was experimenting with trying find the workload clusters on a management, see if any workload clusters are also management clusters, and then recursively probe them for additional workload clusters. I figured I could do that by initializing a client w/ the kubeconfigs stored in the secrets. However, I think that's probably a bit too complicated, and so I'm going to move this logic outside of clusterctl. I think it could be helpful to show more about the relationship between clusters, but we can come back to that at a later time.